digital_asset_types/src/dao/mod.rs

```diff
@@ -88,16 +116,35 @@ impl SearchAssetsQuery {
                     .clone()
                     .map(|x| asset::Column::SpecificationVersion.eq(x)),
             )
+            .add_option(self.token_type.as_ref().map(|x| {
+                match x {
+                    TokenTypeClass::Compressed => asset::Column::TreeId.is_not_null(),
+                    TokenTypeClass::Nft => asset::Column::TreeId.is_null().and(
+                        asset::Column::SpecificationAssetClass
+                            .eq(SpecificationAssetClass::Nft)
+                            .or(asset::Column::SpecificationAssetClass
+                                .eq(SpecificationAssetClass::MplCoreAsset))
+                            .or(asset::Column::SpecificationAssetClass
+                                .eq(SpecificationAssetClass::ProgrammableNft)),
+                    ),
+                    TokenTypeClass::NonFungible => asset::Column::SpecificationAssetClass
+                        .eq(SpecificationAssetClass::Nft)
+                        .or(asset::Column::SpecificationAssetClass
+                            .eq(SpecificationAssetClass::ProgrammableNft))
+                        .or(asset::Column::SpecificationAssetClass
+                            .eq(SpecificationAssetClass::MplCoreAsset)),
```

Comment:
I think for both of these match arms `SpecificationAssetClass::MplCoreCollection` is missing.